src/app/internal/guiapp.cpp

```diff
@@ -232,19 +232,37 @@ void GuiApp::perform()
             m->onDelayedInit();
         }
 
-        startupScenario()->runOnSplashScreen();
-
-        if (splashScreen) {
-            splashScreen->close();
-            delete splashScreen;
+        const Version welcomeDialogLastShownVersion(appshellConfiguration()->welcomeDialogLastShownVersion());
```

Comment:
Can we move all this logic into StartupScenario? 
Ideally, this part should only include method calls like runOnSplashScreen, runAfterSplashScreen, and so on.

Reply:
I would also recommend keeping the methods more abstract, because in your case you’re exposing the internal logic of the scenario.